### PR TITLE
Clarify how GCS client obtain credentials

### PR DESCRIPTION
**Pull request by kthui**
Related PR: https://github.com/triton-inference-server/core/pull/164

### CODE REVIEW

**Comment by GuanLuo** [comment]
Can you add doc on how a user with a public bucket should configure the storage to work with Triton?

**Comment by kthui** [comment]
> Can you add doc on how a user with a public bucket should configure the storage to work with Triton?

Yes, updated.

File: docs/user_guide/model_repository.md [hunk 1]
```diff
@@ -118,7 +118,25 @@ $ tritonserver --model-repository=gs://bucket/path/to/model/repository ...
...
-environment variable should be set.
+environment variable should be set and contains the location of a credential
+JSON file. If no credential is provided, Triton will use credentials from the
+[attached service account](https://cloud.google.com/docs/authentication/application-default-credentials#attached-sa)
+providing a value for the
+[Authorization HTTP header](https://googleapis.dev/cpp/google-cloud-storage/1.42.0/classgoogle_1_1cloud_1_1storage_1_1oauth2_1_1ComputeEngineCredentials.html#a8c3a5d405366523e2f4df06554f0a676) 
+can be obtained. If not obtainable, anonymous credential will be used.
+
+To access buckets with anonymous credential (also known as public bucket), the
+bucket (and objects) should have granted get and list permissions to all users.
```
Comment (Tabrizian) [inline docs/user_guide/model_repository.md:129]: use code block for `get` and `list` words.

**Comment by Tabrizian** [comment]
Is it possible to add testing for GCE credentials?

File: docs/user_guide/model_repository.md (hunk 1 above)
Comment (kthui) [reply docs/user_guide/model_repository.md]: Yes, updated.

**Comment by kthui** [comment]
> Is it possible to add testing for GCE credentials?

It would not be straightforward, but I think it is possible: DLIS-4589.

**Review by Tabrizian — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]

**Pull request merged**